python/engine/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+
+add_custom_command(
+  OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/SWIGPythonRuntime.hxx"
+  COMMAND ${CMAKE_COMMAND} -E env SWIG_LIB="${SWIG_LIB}"
+          "${SWIG_EXECUTABLE}"
+          "-python"
+          -external-runtime "${CMAKE_CURRENT_BINARY_DIR}/SWIGPythonRuntime.hxx"
+)
+
+add_library(
+  pythonengine
+  MODULE
+  PythonEngine.hpp
+  PythonEngine.cpp
+  ${CMAKE_CURRENT_BINARY_DIR}/SWIGPythonRuntime.hxx
+)
+
+file(
+  GENERATE OUTPUT $<TARGET_FILE_DIR:pythonengine>/openstudiodev.py
+  INPUT openstudiodev.py.in
+)
```

Comment:
fancy. configure_file would have done though no? I'm just curious

Reply:
configure_file does not allow generator expressions as far as I'm aware. The whole point of these shenanigans is to handle the configuration subdirectory in multi config builds. file(GENERATE ..) will create one file per config of the target build system. I think this approach is robust.

Full disclaimer I haven't tested on a multi config system yet, but my Windows build is running right now.